src/components/questionary/questionaryComponents/QuestionFormShell.tsx

```diff
@@ -28,82 +29,95 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export const QuestionFormShell = (
-  props: QuestionFormProps & {
-    validationSchema: unknown;
-  }
-) => {
-  const classes = useStyles();
-  const { api } = useDataApiWithFeedback();
-  const definition = getQuestionaryComponentDefinition(props.question.dataType);
+interface QuestionFormShellProps extends QuestionFormProps {
+  validationSchema: unknown;
+  confirm: WithConfirmType;
+}
 
-  const submitHandler = async (values: Question): Promise<void> => {
-    api()
-      .updateQuestion({
-        id: values.id,
-        naturalKey: values.naturalKey,
-        question: values.question,
-        config: values.config ? JSON.stringify(values.config) : undefined,
-      })
-      .then((data) => {
-        if (data.updateQuestion.question) {
-          props.onUpdated?.({
-            ...props.question,
-            ...data.updateQuestion.question,
-          });
-          props.closeMe?.();
-        }
-      });
-  };
+export const QuestionFormShell = withConfirm(
+  (props: QuestionFormShellProps) => {
+    const { question, validationSchema, confirm, children } = props;
+    const { onUpdated, onDeleted, closeMe } = props;
+
+    const classes = useStyles();
+    const { api } = useDataApiWithFeedback();
+    const definition = getQuestionaryComponentDefinition(question.dataType);
 
-  const deleteHandler = () =>
-    api()
-      .deleteQuestion({
-        questionId: props.question.id,
-      })
-      .then((data) => {
-        if (data.deleteQuestion.question) {
-          props.onDeleted?.(data.deleteQuestion.question);
-          props.closeMe?.();
+    const submitHandler = async (values: Question): Promise<void> => {
+      api()
```

Comment:
I personally think that if you use `async/await` here will be a bit cleaner because we reduce the nesting a bit instead of using `.then` with some additional nesting.